azurerm/resource_arm_netapp_account.go

```diff
@@ -0,0 +1,273 @@
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.AccountProperties; props != nil {
+		if err := d.Set("active_directory", flattenArmNetAppActiveDirectories(props.ActiveDirectories)); err != nil {
+			return fmt.Errorf("Error setting `active_directory`: %+v", err)
+		}
+	}
+
+	// Handles tags being interface{} until https://github.com/Azure/azure-rest-api-specs/issues/7447 is fixed
+	currentTags := make(map[string]*string)
+	if v := resp.Tags; v != nil {
+		tagMap := v.(map[string]interface{})
+
+		for k, v := range tagMap {
+			currentTags[k] = utils.String(v.(string))
+		}
+	}
+
+	return tags.FlattenAndSet(d, currentTags)
+}
+
+func resourceArmNetAppAccountDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.AccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["netAppAccounts"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting NetApp Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting NetApp Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmNetAppActiveDirectories(input []interface{}) *[]netapp.ActiveDirectory {
+	results := make([]netapp.ActiveDirectory, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		dns := strings.Join(*utils.ExpandStringSlice(v["dns_servers"].([]interface{})), ",")
+
+		result := netapp.ActiveDirectory{
+			DNS:                utils.String(dns),
+			Domain:             utils.String(v["domain"].(string)),
+			OrganizationalUnit: utils.String(v["organizational_unit"].(string)),
+			Password:           utils.String(v["password"].(string)),
+			SmbServerName:      utils.String(v["smb_server_name"].(string)),
+			Username:           utils.String(v["username"].(string)),
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func flattenArmNetAppActiveDirectories(input *[]netapp.ActiveDirectory) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		b := make(map[string]interface{})
+
+		if v := item.DNS; v != nil {
+			dnsServers := strings.Split(*v, ",")
+			b["dns_servers"] = utils.FlattenStringSlice(&dnsServers)
+		}
+		if v := item.Domain; v != nil {
+			b["domain"] = *v
+		}
+		if v := item.SmbServerName; v != nil {
+			b["smb_server_name"] = *v
+		}
+		if v := item.Username; v != nil {
+			b["username"] = *v
+		}
+		if v := item.Password; v != nil {
+			b["password"] = *v
+		}
```

Comment:
here, what we should be doing is something along the lines of:
```
if v, ok := d.GetOk("active_directory.0.password"); ok {
    b["password"] = v
}
```

Reply:
Fixed